### PR TITLE
refactor: replace `PRSocialCard` with githubs Opengraph image

### LINKED ISSUE #883: feat: use opengraph images for highlights
Today the highlights card is presented as a custom UI element. We should be fetching directly from the OpenGraph url. This enables us to display images for more than just GitHub. But for now, this should displayed as an image and not customer elements. 

```
img src="https://opengraph.githubassets.com/1/open-sauced/hot/pull/448"
```

Not that any github image url will render an OG image using the prefix `https://opengraph.githubassets.com/1/` and adding the url minus the `https://github.com/`

### PR DESCRIPTION
**Pull request by OgDev-01**
## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description
This PR:

- refactors the highlight social image to leverage GitHub [open graph image](https://github.blog/2021-06-22-framework-building-open-graph-images/)

## Related Tickets & Documents
Fixes #883 

## Mobile & Desktop Screenshots/Recordings
image

## Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by bdougie** [comment]
mobile width/height needs to be address. image
<details><summary>Browser metadata</summary>





```
Path:      /user/bdougie
Browser:   Chrome 109.0.0.0 on Mac OS 10.15.7
Viewport:  559 x 830 @2x
Language:  en-US
Cookies:   Enabled
```

[Open in BrowserStack](https://www.browserstack.com/user/try-live?url=https%3A%2F%2Fdeploy-preview-889--oss-insights.netlify.app%2Fuser%2Fbdougie&os=OS%20X&os_version=Catalina&browser=Chrome&browser_version=109.0&resolution=559x830&speed=1&start=true&ref=netlify-source)



</details>

[Open Deploy Preview](https://deploy-preview-889--oss-insights.netlify.app/user/bdougie) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=63f3d0c24b92c7000879a4c3&commentID=63f3d19d8f9dfb26c606d66a&resolution=resolved)

File: lib/utils/github.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ const generateApiPrUrl = (
     const trimmedUrl = url.trim();
     const githubUrl = new URL(trimmedUrl.includes("https://") ? trimmedUrl : `https://${trimmedUrl}`);
     const { pathname } = githubUrl;
-
+    console.log(pathname);
```
Comment (0-vortex) [inline lib/utils/github.ts:23]: stray

File: next.config.js [hunk 2]
```diff
@@ -7,7 +7,8 @@ module.exports = {
       "images.unsplash.com",
       "www.github.com",
       "github.com",
-      "res.cloudinary.com"
+      "res.cloudinary.com",
+      "opengraph.githubassets.com"
```
Comment (0-vortex) [inline next.config.js:11]: don't optimise github images: https://nextjs.org/docs/basic-features/image-optimization#domains

Comment (OgDev-01) [reply next.config.js:11]: Ok

**Comment by OgDev-01** [comment]
> mobile width/height needs to be address. image
> 
> Browser metadata
> [Open Deploy Preview](https://deploy-preview-889--oss-insights.netlify.app/user/bdougie) · [Mark as Resolved](https://app.netlify.com/cdp/resolve?deployID=63f3d0c24b92c7000879a4c3&commentID=63f3d19d8f9dfb26c606d66a&resolution=resolved)

Fixed

**Review by bdougie — APPROVED** [review]

File: components/molecules/GhOpenGraphImg/gh-open-graph-img.tsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+interface GhOpenGraphImgProps {
+  githubLink: string;
+}
+
+const GhOpenGraphImg = ({ githubLink }: GhOpenGraphImgProps): JSX.Element => {
+  const { isValid, url } = generateGhOgImage(githubLink);
+
+  return (
+    // eslint-disable-next-line @next/next/no-img-element
```
Comment (0-vortex) [inline components/molecules/GhOpenGraphImg/gh-open-graph-img.tsx:14]: please stop disabling eslint rules and state your problems

Comment (OgDev-01) [reply components/molecules/GhOpenGraphImg/gh-open-graph-img.tsx:14]: It's a nextjs warning to not use the native `img` tag in next https://nextjs.org/docs/messages/no-img-element

Comment (bdougie) [reply components/molecules/GhOpenGraphImg/gh-open-graph-img.tsx:14]: That seems aggressive for linter, but disabling it seems like a band-aid. Are there other alternatives @OgDev-01?

**Review by brandonroberts — APPROVED** [review]
LGTM. Need to resolve solution on lint rule

File: components/molecules/GhOpenGraphImg/gh-open-graph-img.tsx (hunk 3 above)
Comment (OgDev-01) [reply components/molecules/GhOpenGraphImg/gh-open-graph-img.tsx:14]: Another alternative is to disable globally in the `.eslintrc` rules
image

**Pull request merged**